### DISCUSSION TITLE
Otterkit's COBOL frontend is now on hiatus for an indeterminate amount of time.

### DISCUSSION THREAD

**Discussion by KTSnowy** [otterkit/otterkit-cobol#42] [General Discussion]
I'm opening this as an organization discussion rather than a repository issue because it feels more appropriate. 

Just as the title says, the COBOL frontend will now be on hiatus for an unknown amount of time. I'll be archiving the frontend code for now, maybe in the future we can revisit the idea of building the COBOL frontend again but for now I cannot justify continuing to work on it.

I'm not abandoning the project entirely though, specially not the "Otterkit" name because it's a pretty good name and we already own the otterkit.com domain. It's difficult to find a good project name today, and much more difficult to find a good .com domain available for it. Otterkit will continue to be a compiler project, just not a _COBOL_ compiler project. The project will continue, but with a different programming language.

@gabrielesilinic and @TriAttack238, I have deleted my discord account, I'm not sure how long it will take for them to show that it was deleted. We should discuss about the project here on GitHub, publicly. Having project discussions privately on discord seems like a really bad idea from an open source perspective, so I'm taking the nuclear option and I'm deleting my discord account. Further discussions about the project should happen here (organization discussions) or on repository issues from now on.

For now I probably should disappear from all social media until I have something useful to show, because I've been talking too much and not showing enough code. I need to stop talking and start writing more code.

**Comment by gabrielesilinic** [comment]
> because I've been talking too much and not showing enough code. 

I'd rather say the COBOL compiler project started to get too complex compared to what we could actually do, I'd rather suggest doing a pet project without too much presumptions and refining it with time in a manner that may later on help us let any language sit on top of it, a bit like the JVM.

I'd focus on the web first, webassembly in particular as it may be a good technology to help us get popular in any shape or form since you literally need just a browser (most people already have) to get started. I'd suggest to also follow the trend of hosting webassembly in other places, this at least until we have any resources to build a native compiler.

It may be a full stack framework in the mix or not, but in any case it's just a way to get ready to go back to COBOL someday.

**Reply by KTSnowy** [reply]
> I'd rather suggest doing a pet project without too much presumptions and refining it with time in a manner that may later on help us let any language sit on top of it, a bit like the JVM.

I'm working on a design for something similar to that right now. I won't talk too much about the design here because I don't have working code to back it up. I'll talk about it once I actually have something working.

I might also change the otterkit repo's description and readme to "Otterkit Toolchain" rather than the "Otterkit COBOL Compiler" we have right now.

**Reply by gabrielesilinic** [reply]
I still have the https://otterk.it short domain, let me know if you want it, it still points to https://otterkit.com

**Reply by KTSnowy** [reply]
We could try making a domain transfer, does the registrar you registered the domain with allow domain transfers?

**Reply by KTSnowy** [reply]
@gabrielesilinic we might not be able to transfer it I think. Cloudflare doesn't support the .it extension.

**Comment by TriAttack238** [comment]
That is unfortunate, but understandable. I will continue to develop my skills, and maybe one day we'll meet again to make something new. Best of luck on your future endeavors @KTSnowy @gabrielesilinic! And don't be afraid to let me know if you have any new projects!
